lib/core/macrocache.nim

```diff
@@ -7,38 +7,199 @@
 #    distribution, for details about the copyright.
 #
 
-## This module provides an API for macros that need to collect compile
-## time information across module boundaries in global variables.
-## Starting with version 0.19 of Nim this is not directly supported anymore
-## as it breaks incremental compilations.
-## Instead the API here needs to be used.
+## This module provides an API for macros to collect compile-time information
```

Comment:
> One thing which I'm still not sure of is how we can show the usage in different modules.

we should implement https://github.com/timotheecour/Nim/issues/637 (HAR) which would allow:

```nim
runnableExamples(multifile = true):
  #### HAR
  #### fn1.nim
  const mcTable = CacheTable"myTable"
  mcTable["foo"] = newLit("a")
  #### fn2.nim
  const mcTable = CacheTable"myTable"
  echo mcTable["foo"]
```

obviously, this should be done in future work and not in this PR :)